src/Timelock.sol

```diff
@@ -0,0 +1,739 @@
+    /// Note that a "pending" operation may also be "ready".
+    function isOperationPending(bytes32 id) public view virtual returns (bool) {
+        return getTimestamp(id) > _DONE_TIMESTAMP;
+    }
+
+    /// @dev Returns whether an operation is ready for execution.
+    /// Note that a "ready" operation is also "pending".
+    /// cannot be executed after the expiry period.
+    function isOperationReady(bytes32 id) public view virtual returns (bool) {
+        uint256 timestamp = getTimestamp(id);
+        return
+            timestamp > _DONE_TIMESTAMP &&
+            timestamp <= block.timestamp &&
+            timestamp + expirationPeriod > block.timestamp;
+    }
+
+    /// @dev Returns whether an operation is done or not.
+    function isOperationDone(bytes32 id) public view virtual returns (bool) {
+        return getTimestamp(id) == _DONE_TIMESTAMP;
+    }
+
+    /// @dev Returns the timestamp at which an operation becomes ready (0 for
+    /// unset operations, 1 for done operations).
+    function getTimestamp(bytes32 id) public view virtual returns (uint256) {
+        return timestamps[id];
+    }
+
+    /// @dev Returns the identifier of an operation containing a single transaction.
+    function hashOperation(
+        address target,
+        uint256 value,
+        bytes calldata data,
+        bytes32 predecessor,
+        bytes32 salt
+    ) public pure virtual returns (bytes32) {
+        return keccak256(abi.encode(target, value, data, predecessor, salt));
+    }
+
+    /// @dev Returns the identifier of an operation containing a batch of transactions.
+    function hashOperationBatch(
+        address[] calldata targets,
+        uint256[] calldata values,
+        bytes[] calldata payloads,
+        bytes32 predecessor,
+        bytes32 salt
+    ) public pure virtual returns (bytes32) {
+        return
+            keccak256(abi.encode(targets, values, payloads, predecessor, salt));
+    }
+
+    /// ---------------------------------------------------------------
+    /// ---------------------------------------------------------------
+    /// -------------------- Timelock Functions -----------------------
+    /// ---------------------------------------------------------------
+    /// ---------------------------------------------------------------
+
+    /// @dev Schedule an operation containing a single transaction.
+    /// Emits {CallSalt} if salt is nonzero, and {CallScheduled}.
+    /// Requirements:
+    ///   the caller must have the 'proposer' role.
+    function schedule(
+        address target,
+        uint256 value,
+        bytes calldata data,
+        bytes32 predecessor,
+        bytes32 salt,
+        uint256 delay
+    ) public virtual onlySafe whenNotPaused {
+        bytes32 id = hashOperation(target, value, data, predecessor, salt);
+
+        require(_liveProposals.add(id), "failed to add proposal, duplicate id");
+
+        _schedule(id, delay);
```

Comment:
why no call to checkCalldata here?

Reply:
I guess because any calldata can be executed as long is using the timelock delay?